src/gui/speciesEditor.cpp

```diff
@@ -259,3 +260,34 @@ void SpeciesEditor::on_ToolsMinimiseButton_clicked(bool checked)
     speciesViewer()->postRedisplay();
     speciesViewer()->notifyDataModified();
 }
+
+void SpeciesEditor::on_ToolsAddCellButton_clicked(bool checked)
+{
+    SpeciesCellDialog dialog(this, speciesViewer()->species());
+
+    if (dialog.createUnitCell())
+    {
+        speciesViewer()->species()->recalculateIntermolecularTerms(tolerance_);
+        speciesViewer()->view().showAllData();
+        speciesViewer()->postRedisplay();
+        speciesViewer()->notifyDataModified();
+    }
+}
+
+void SpeciesEditor::on_ToolsRemoveCellButton_clicked(bool checked)
+{
+    speciesViewer()->species()->recalculateIntermolecularTerms(tolerance_);
+    speciesViewer()->species()->removeBox();
+    speciesViewer()->view().showAllData();
+    speciesViewer()->postRedisplay();
+    speciesViewer()->notifyDataModified();
+}
+
+void SpeciesEditor::on_ToolsBondToleranceSpin_valueChanged(double value)
+{
+    if (tolerance_ != value) // Segfault without
+    {
+        tolerance_ = value;
+        speciesViewer()->species()->recalculateIntermolecularTerms(tolerance_);
+    }
```

Comment:
Just need to protect against the widget value being set when there is no current species, as other functions do.  You can also do away with the `tolerance_` local and just have the default set to 1.1 in the UI file.  Have also added a call to `postRedisplay()` so the view updates when you change the bond tolerance.
```suggestion
    // Get displayed Species
    const auto *sp = speciesViewer()->species();
    if (!sp)
        return;

    speciesViewer()->species()->recalculateIntermolecularTerms(value);
    speciesViewer()->postRedisplay();
```